src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -997,6 +997,16 @@ public ErrorDiagnostic VariableLoopsRuntimeDependencyNotAllowed(IEnumerable<stri
                     "BCP175",
                     $"The variable for-expression body or array expression must be evaluable at the start of the deployment and cannot depend on any values that have not yet been calculated.{variableDependencyChainClause}");
             }
+
+            public ErrorDiagnostic FunctionOnlyValidInModuleParameterAssignment(string functionName) => new(
+                TextSpan,
+                "BCP176",
+                $"Function \"{functionName}\" is not valid at this location. It can only be used in assigning value to a module parameter.");
```

Comment:
nit - Can we tweak the second sentence:

"It can only be used when assigning a value to a module parameter"